zipkin-local.js

```diff
@@ -0,0 +1,148 @@
+const _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
+const {
+  PLUGIN_OPTIONS,
+} = require("gatsby-plugin-newrelic-test/utils/constants");
+const getCiData = require("./utils/getCiData");
+const _zipkin = _interopRequireDefault(require("zipkin"));
+const _zipkinTransportHttp = require("zipkin-transport-newrelic");
+const _zipkinJavascriptOpentracing = _interopRequireDefault(
+  require("zipkin-javascript-opentracing")
+);
+const _nodeFetch = _interopRequireDefault(require("node-fetch"));
+const {now} = require('./utils/time');
+
+const ciAttributes = getCiData();
+
+let recorder, 
+    logger;
+/**
+ * Create and return an open-tracing compatible tracer. See
+ * https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.ts
+ */
+
+const create = () => {
+  const { staging, NR_INGEST_KEY, traces: tags = {}, SITE_NAME } = PLUGIN_OPTIONS;
+  logger = new _zipkinTransportHttp.HttpLogger({
+    endpoint: `https://${
+      staging ? `staging-` : ``
+    }trace-api.newrelic.com/trace/v1`,
+    jsonEncoder: _zipkin.jsonEncoder.JSON_V2,
+    httpInterval: 2000,
+    headers: {
+      "Api-Key": NR_INGEST_KEY,
+      "Data-Format": "zipkin",
+      "Data-Format-Version": 2,
+      options: PLUGIN_OPTIONS,
+      tags: {
+        ...tags,
+        ...ciAttributes,
+      },
+    },
+  });
+  recorder = new _zipkin.BatchRecorder({
+    logger,
+    // timeout = 60 hours, must be longer than site's build time
+    timeout: 60 * 60 * 60 * 1000000,
+  });
+  const tracer = new _zipkinJavascriptOpentracing.default({
+    localServiceName: SITE_NAME,
+    serviceName: SITE_NAME,
+    sampler: new _zipkin.sampler.CountingSampler(1),
+    traceId128Bit: true,
+    supportsJoin: true,
+    recorder,
+    kind: `client`,
+  });
+  return tracer;
+};
+
+const formatTrace = (trace) => {
+  const { SITE_NAME, traces: {tags} = {}, buildId } = PLUGIN_OPTIONS;
+  trace = JSON.parse(trace);
+  trace.localEndpoint = {};
+  trace.localEndpoint.serviceName = SITE_NAME;
+  if(!trace.tags) {
+    trace.tags = {};
+  }
+  if (trace.annotations) {
+    for (let anno of trace.annotations) {
+      trace.tags[anno.key] = anno.value;
+    }
+  }
+  if (trace.binaryAnnotations) {
+    for (let anno of trace.binaryAnnotations) {
+      trace.tags[anno.key] = anno.value;
+    }
+  }
+  if (trace.name === "run-api") {
+    trace.name += `: ${trace.tags.api}`;
+  }
+  if (trace.name === "run-plugin") {
+    trace.name += `: ${trace.tags.plugin}`;
+  }
+  trace.tags = {...trace.tags, ...tags};
+  trace.tags.thing = 'zipkin-local';
+  trace.tags.buildId = buildId;
+  return JSON.stringify(trace);
+}
+
+const sendTraceQueue = async (queue) => {
+  const { NR_INGEST_KEY } = PLUGIN_OPTIONS;
+  const response = await (0, _nodeFetch.default)(logger.endpoint, {
```

Comment:
Out of curiosity, why are we writing this async call this way with the `(0, _nodefetch.default)` stuff in the front? Could we just do:

```js
const response = await fetch(logger.endpoint, {
  // ..options
});
```